File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>(); // val : idx

        int i = 0;
        for (int n : nums) {
            int diff = target - n;
            if (map.containsKey(diff)) {
                int[] idx = {map.get(diff), i};
                return idx;
            }
            map.put(n, i);
            i++;
        }

        return null;
    }
}

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        '''
        [3,4,5,6,1,2]
        target = 1

        '''
        for i in range(len(nums)):
            if nums[i] == target:
                return i
        return -1